util/db/repository_secrets.go

```diff
@@ -0,0 +1,439 @@
+		return err
+	}
+
+	return s.db.settingsMgr.ResyncInformers()
+}
+
+func (s *secretsRepositoryBackend) RepoCredsExists(ctx context.Context, repoURL string) (bool, error) {
+	_, err := s.getRepoCredsSecret(repoURL)
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (s *secretsRepositoryBackend) GetAllHelmRepoCreds(ctx context.Context) ([]*appsv1.RepoCreds, error) {
+	var helmRepoCreds []*appsv1.RepoCreds
+
+	secrets, err := s.db.listSecretsByType(common.LabelValueSecretTypeRepoCreds)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range secrets {
+		if strings.EqualFold(string(secret.Data["type"]), "helm") {
+			repoCreds, err := s.secretToRepoCred(secret)
+			if err != nil {
+				return nil, err
+			}
+
+			helmRepoCreds = append(helmRepoCreds, repoCreds)
+		}
+	}
+
+	return helmRepoCreds, nil
+}
+
+func (s *secretsRepositoryBackend) secretToRepository(secret *corev1.Secret) (*appsv1.Repository, error) {
+	repository := &appsv1.Repository{
+		Name:                       string(secret.Data["name"]),
+		Repo:                       string(secret.Data["url"]),
+		Username:                   string(secret.Data["username"]),
+		Password:                   string(secret.Data["password"]),
+		SSHPrivateKey:              string(secret.Data["sshPrivateKey"]),
+		TLSClientCertData:          string(secret.Data["tlsClientCertData"]),
+		TLSClientCertKey:           string(secret.Data["tlsClientCertKey"]),
+		Type:                       string(secret.Data["type"]),
+		GithubAppPrivateKey:        string(secret.Data["githubAppPrivateKey"]),
+		GitHubAppEnterpriseBaseURL: string(secret.Data["githubAppEnterpriseBaseUrl"]),
+		Proxy:                      string(secret.Data["proxy"]),
+	}
+
+	insecureIgnoreHostKey, err := boolOrFalse(secret, "insecureIgnoreHostKey")
+	if err != nil {
+		return repository, err
+	}
+	repository.InsecureIgnoreHostKey = insecureIgnoreHostKey
+
+	insecure, err := boolOrFalse(secret, "insecure")
+	if err != nil {
+		return repository, err
+	}
+	repository.Insecure = insecure
+
+	enableLfs, err := boolOrFalse(secret, "enableLfs")
+	if err != nil {
+		return repository, err
+	}
+	repository.EnableLFS = enableLfs
+
+	enableOCI, err := boolOrFalse(secret, "enableOCI")
+	if err != nil {
+		return repository, err
+	}
+	repository.EnableOCI = enableOCI
+
+	githubAppID, err := intOrZero(secret, "githubAppID")
+	if err != nil {
+		return repository, err
+	}
+	repository.GithubAppId = githubAppID
+
+	githubAppInstallationID, err := intOrZero(secret, "githubAppInstallationID")
+	if err != nil {
+		return repository, err
+	}
+	repository.GithubAppInstallationId = githubAppInstallationID
+
+	return repository, nil
+}
+
+func (s *secretsRepositoryBackend) repositoryToSecret(repository *appsv1.Repository, secret *corev1.Secret) {
+	data := make(map[string][]byte)
+
+	data["name"] = []byte(repository.Name)
```

Comment:
Please don't add keys with empty values to the data, so that secret store only values specified by user.

Reply:
Good catch, I have not thought about that! Thanks for pointing that out!